modules/ROOT/pages/constraints/managing-constraints.adoc

```diff
@@ -0,0 +1,1598 @@
+It also ensures that all properties in the set are present.
+
+.Create a constraint requiring `KNOWS` relationships to have a unique combination of `since` and `how` properties as a relationship key
+[source, cypher]
+----
+CREATE CONSTRAINT knows_since_how
+FOR ()-[knows:KNOWS]-() REQUIRE (knows.since, knows.how) IS RELATIONSHIP KEY
+----
+
+.Result
+[source, queryresult]
+----
+Added 1 constraint.
+----
+
+======
+
+[[create-key-constraint-compliant-data]]
+==== Create data that complies with existing key constraints
+
+.Creating a node that complies with existing node key constraints
+======
+
+.Create an `Actor` node with `firstname` and `surname` properties
+[source, cypher]
+----
+CREATE (actor:Actor {firstname: 'Keanu', surname: 'Reeves'})
+----
+
+.Result
+[source, queryresult]
+----
+Added 1 label, created 1 node, set 2 properties.
+----
+
+======
+
+
+.Creating a relationship that complies with existing relationship key constraints
+======
+
+.Create a `KNOWS` relationship with both `since` and `how` properties
+[source, cypher]
+----
+CREATE (:Actor {firstname: 'Jensen', surname: 'Ackles'})-[:KNOWS {since: 2008, how: 'coworkers', friend: true}]->(:Actor {firstname: 'Misha', surname: 'Collins'})
+----
+
+.Result
+[source, queryresult]
+----
+Added 2 labels, created 2 nodes, set 6 properties, created 1 relationship.
+----
+
+======
+
+
+[role=label--new-5.16]
+=== Create a constraint with a parameter
+
+All constraint types can be created with a parameterized name.
+
+.Create a node property uniqueness constraint using a parameter
+======
+
+.Parameters
+[source, parameters]
+----
+{
+  "name": "node_uniqueness_param"
+}
+----
+
+.Create a node property uniqueness constraint with a parameterized name
+[source, cypher]
+----
+CREATE CONSTRAINT $name
+FOR (book:Book) REQUIRE book.prop1 IS UNIQUE
+----
+
+.Result
+[source, queryresult]
+----
+Added 1 constraint.
+----
+
+======
+
+.Create a relationship property existence constraint using a parameter
+======
+
+.Parameters
+[source, parameters]
+----
+{
+  "name": "rel_exist_param"
+}
+----
+
+.Create a relationship property existence constraint with a parameterized name
+[source, cypher]
+----
+CREATE CONSTRAINT $name
+FOR ()-[wrote:WROTE]-() REQUIRE wrote.published IS NOT NULL
+----
+
+.Result
+[source, queryresult]
+----
+Added 1 constraint.
+----
+
+======
+
+[[multiple-constrains]]
+=== Multiple constraints on the same property combinations
+
+Some constraint types are allowed on the same label/relationship type and property combination.
+For example, it is possible to have a uniqueness and an existence constraint on the same label/relationship type and property combination, though this would be the equivalent of having a node or relationship key constraint.
+A more useful example would be to combine a property type and an existence constraint to ensure that the property exists and has the given type.
+
+[[constraints-and-indexes]]
+=== Constraints and backing indexes
```

Comment:
for all of these common sections, do we want labels on when different things were introduced or do we leave that to just the relevant parts above? (such as the examples on relationship property uniqueness and key constraints)

Reply:
I think we decided we didn't need it in general, but just in the relevant create sections where they first appear?

Reply:
Decision: only needed when introduced